beluga/include/beluga/circular_array.hpp

```diff
@@ -0,0 +1,539 @@
+    }
+  }
+
+ private:
+  // Computes the head index of the array.
+  [[nodiscard]] constexpr size_type head_index() const noexcept {
+    return static_cast<size_type>(size_ > tail_index_ + 1) * N + tail_index_ - size_ + 1;
+  }
+
+  std::array<T, N> data_;
+  size_type tail_index_{0U};
+  size_type size_{0U};
+};
+
+/// Convenient type alias for a circular array that behaves like a rolling window.
+/**
+ * A rolling window automatically overwrites older values, it extrapolates its last
+ * value to always seem full, and it is accessed by the front.
+ */
+template <typename T, std::size_t N>
+using RollingWindow = CircularArray<
+    T,
+    N,
+    CircularArrayFeatureFlags::kRolloverOnWrite | CircularArrayFeatureFlags::kExtrapolateOnRead |
+        CircularArrayFeatureFlags::kLayoutReversal>;
+
+/// Convenient stream operator overload to push a `value` to a circular `array`.
+/**
+ * Functionally equivalent to a push_back() (or push_front() if the layout
+ * reversal feature is enabled).
+ */
+template <typename T, std::size_t N, CircularArrayFeatureFlags F>
+CircularArray<T, N, F>& operator<<(CircularArray<T, N, F>& array, T value) {
+  if constexpr (F & CircularArrayFeatureFlags::kLayoutReversal) {
+    array.push_front(std::move(value));
+  } else {
+    array.push_back(std::move(value));
+  }
+  return array;
+}
+
+/// Gets an lvalue reference to the ith value in a given `array`.
+/**
+ * This is an `std::get` overload that relies on argument-dependent lookup (ADL).
+ */
+template <std::size_t I, class T, std::size_t N, CircularArrayFeatureFlags F>
+constexpr T& get(CircularArray<T, N, F>& array) noexcept {
+  return array[I];
+}
+
+/// Gets an rvalue reference to the ith value in a given `array`.
+/**
+ * This is an `std::get` overload that relies on argument-dependent lookup (ADL).
+ */
+template <std::size_t I, class T, std::size_t N, CircularArrayFeatureFlags F>
+constexpr T&& get(CircularArray<T, N, F>&& array) noexcept {
+  return array[I];
+}
+
+/// Gets a constant lvalue reference to the ith value in a given `array`.
+/**
+ * This is an `std::get` overload that relies on argument-dependent lookup (ADL).
+ */
+template <std::size_t I, class T, std::size_t N, CircularArrayFeatureFlags F>
+constexpr const T& get(const CircularArray<T, N, F>& array) noexcept {
+  return array[I];
+}
+
+/// Gets a constant rvalue reference to the ith value in a given `array`.
+/**
+ * This is an `std::get` overload that relies on argument-dependent lookup (ADL).
+ */
+template <std::size_t I, class T, std::size_t N, CircularArrayFeatureFlags F>
+constexpr const T&& get(const CircularArray<T, N, F>&& array) noexcept {
+  return array[I];
+}
+
+/// Swaps arrays `a` and `b`.
+/**
+ * This is an `std::swap` overload that relies on argument-dependent lookup (ADL).
+ */
+template <typename T, std::size_t N, CircularArrayFeatureFlags F, CircularArrayFeatureFlags G>
+constexpr void swap(CircularArray<T, N, F>& a, CircularArray<T, N, G>& b) {
+  a.swap(b);
+};
```

Comment:
The linter is complaining about this.

```suggestion
}
```

Reply:
Yes! Fixed in https://github.com/Ekumen-OS/beluga/pull/320/commits/e1e3e0684cce1d489e0d87b469a2afa86f960477.